Flaky test

```
    @pytest.mark.parametrize('live_update', [True, False])
    def test_drag_resize_floating_client(hlwm, x11, mouse, live_update):
        hlwm.call(['set', 'update_dragged_clients', hlwm.bool(live_update)])
    
        client, winid = x11.create_client(geometry=(50, 50, 300, 200))
        hlwm.call(f'set_attr clients.{winid}.floating true')
        geom_before = client.get_geometry()
        x_before, y_before = x11.get_absolute_top_left(client)
        assert (geom_before.width, geom_before.height) == (300, 200)
        # move cursor to the top left corner, so we change the
        # window position and the size (and the bottom right corner is fixed)
        mouse.move_into(winid, x=0, y=0)
    
        hlwm.call(['drag', winid, 'resize'])
        assert hlwm.get_attr('clients.dragged.winid') == winid
        mouse.move_relative(100, 120)
        final_size = (geom_before.width - 100, geom_before.height - 120)
    
        # check geometry during drag
        geom_after = client.get_geometry()
        x_after, y_after = x11.get_absolute_top_left(client)
        assert (x_after, y_after) == (x_before + 100, y_before + 120)
        expected_size = (geom_before.width, geom_before.height)
        if live_update:
            expected_size = final_size
>       assert (geom_after.width, geom_after.height) == expected_size
E       assert (300, 200) == (200, 80)
E         At index 0 diff: 300 != 200
E         Full diff:
E         - (200, 80)
E         + (300, 200)

```

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -838,7 +838,7 @@ def keyboard():
 
 
 @pytest.fixture()
-def mouse(hlwm_process):
+def mouse(hlwm_process, hlwm):
     class Mouse:
         def move_into(self, win_id, x=1, y=1):
             self.call_cmd(f'xdotool mousemove --sync --window {win_id} {x} {y}', shell=True)
@@ -849,22 +849,38 @@ def mouse(hlwm_process):
             if wait:
                 with hlwm_process.wait_stderr_match('ButtonPress'):
                     subprocess.check_call(['xdotool', 'click', button])
+                # reaching this line only means that hlwm started processing
+                # the ButtonPress. So we need to wait until the event is fully processed:
+                hlwm.call('true')
             else:
                 subprocess.check_call(['xdotool', 'click', button])
 
-        def move_to(self, abs_x, abs_y):
+        def move_to(self, abs_x, abs_y, wait=True):
             abs_x = str(int(abs_x))
             abs_y = str(int(abs_y))
             self.call_cmd(f'xdotool mousemove --sync {abs_x} {abs_y}', shell=True)
+            if wait:
+                # wait until all the mouse move events that are now in the queue
+                # are fully processed:
+                hlwm.call('true')
 
-        def move_relative(self, delta_x, delta_y):
+        def move_relative(self, delta_x, delta_y, wait=True):
             self.call_cmd(f'xdotool mousemove_relative --sync {delta_x} {delta_y}', shell=True)
+            if wait:
+                # wait until all the mouse move events that were put in the
+                # queue by the above xdotool invokation are fully processed.
+                # (other than for the button events, the motion notify events
+                # are not printed to stderr, because this would lead to
+                # to much debug output on motions of the physical mouse)
+                hlwm.call('true')
 
         def mouse_press(self, button, wait=True):
             cmd = ['xdotool', 'mousedown', button]
             if wait:
                 with hlwm_process.wait_stderr_match('ButtonPress'):
                     subprocess.check_call(cmd)
+                # wait for the ButtonPress to be fully processed:
+                hlwm.call('true')
             else:
                 subprocess.check_call(cmd)
 
@@ -873,6 +889,8 @@ def mouse(hlwm_process):
             if wait:
                 with hlwm_process.wait_stderr_match('ButtonRelease'):
                     subprocess.check_call(cmd)
+                # wait for the ButtonRelease to be processed
+                hlwm.call('true')
             else:
                 subprocess.check_call(cmd)
 

--- a/tests/test_mousebind.py
+++ b/tests/test_mousebind.py
@@ -191,6 +191,7 @@ def test_drag_resize_floating_client(hlwm, x11, mouse, live_update):
     final_size = (geom_before.width - 100, geom_before.height - 120)
 
     # check geometry during drag
+    x11.display.sync()
     geom_after = client.get_geometry()
     x_after, y_after = x11.get_absolute_top_left(client)
     assert (x_after, y_after) == (x_before + 100, y_before + 120)

